### PR TITLE
[DEV-2447] Implement catalog soft delete

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

Add endpoint to support catalog soft delete.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: Taskfile.yaml [hunk 1]
```diff
@@ -176,7 +176,7 @@ tasks:
       - featurebyte/**/*
     cmds:
       - poetry run pip-licenses --packages '{{ .PACKAGES }}' --allow-only='{{ .PERMISSIVE_LICENSES }}'
-      - poetry run pip-audit --ignore-vuln PYSEC-2023-177
+      - poetry run pip-audit --ignore-vuln PYSEC-2023-177 --ignore-vuln GHSA-g4mx-q9vg-27p4
```
Comment (YS-L) [inline Taskfile.yaml:179]: Shouldn't be needed once https://github.com/featurebyte/featurebyte/pull/1830 is merged.

Comment (gee-senbong) [reply Taskfile.yaml:179]: Will update this.

File: featurebyte/routes/catalog/api.py [hunk 2]
```diff
@@ -41,8 +41,9 @@ class CatalogRouter(BaseApiRouter[CatalogModel, CatalogList, CatalogCreate, Cata
 
     def __init__(self) -> None:
         super().__init__("/catalog")
+        self.remove_routes({"/catalog/{catalog_id}": ["DELETE"]})
```
[the remark this answers is not in the document]
Comment (gee-senbong) [reply featurebyte/routes/catalog/api.py:44]: I am not sure whether appending a new delete route will override existing one. That's why I added an explicit statement to remove existing delete route.

**Review by YS-L — APPROVED** [review]

File: Taskfile.yaml (hunk 1 above)
Comment (gee-senbong) [reply Taskfile.yaml:179]: Will remove the ignore vuln in later PR (likely in https://github.com/featurebyte/featurebyte/pull/1831) once https://github.com/featurebyte/featurebyte/pull/1830 is merged.

**Pull request merged**